### PR TITLE
Implement equals and hashcode for UnsolvedClass, removing another infinitely-looping test case

### PR DESCRIPTION
**Pull request by kelloggm**
...but #87 still runs forever

### CODE REVIEW

File: src/main/java/org/checkerframework/specimin/UnsolvedClass.java [hunk 1]
```diff
@@ -204,6 +206,24 @@ public void updateFieldByType(String currentType, String correctType) {
     }
   }
 
+  @Override
+  public boolean equals(@Nullable Object other) {
+    if (!(other instanceof UnsolvedClass)) {
+      return false;
+    }
+    UnsolvedClass otherClass = (UnsolvedClass) other;
+    // This makes me nervous, because it's not clear that it
```
Comment (LoiNguyenCS) [inline src/main/java/org/checkerframework/specimin/UnsolvedClass.java:215]: An `UnsolvedClass` instance represents a synthetic class to be created by Specimin. And we don't create anonymous synthetic classes, so this case will not happen.

**Review by LoiNguyenCS — CHANGES_REQUESTED** [review]
There is just one minor comment that I think we should remove. Other than that, this PR looks good to me.

File: src/main/java/org/checkerframework/specimin/UnsolvedClass.java (hunk 1 above)
Comment (kelloggm) [reply src/main/java/org/checkerframework/specimin/UnsolvedClass.java]: Thanks for correcting this. I wrote a comment there instead to clarify why this is safe.

**Review by LoiNguyenCS — APPROVED** [review]
LGTM

**Pull request merged**